### PR TITLE
Agent card tweak

### PR DESCRIPTION
**Pull request by Yenwodyah**
## About The Pull Request

Allows agent cards to be forged by anyone if they're used in an unforged state by a non-traitor. Previously, to prevent metagaming, only antags could forge agent IDs.

## Why It's Good For The Game

There's no reason to hide the fact that a card is an agent ID if it hasn't been forged into anything yet. Now if a non-antag gets an unforged agent ID from null crates, stolen uplinks, captured traitors, etc. they can use it to call themself bogdanoff and yell "dump eet" over comms. And other hilarious memes.

## Changelog
:cl: Yenwodyah
tweak: Non-antags can now forge all agent IDs if they find them before they've been forged.
/:cl:

### CODE REVIEW

**Comment by imsxz** [comment]
the point of it was to prevent sec from using agent ID cards in hand to check if it was an agent ID card, the issue was that nonantags still got the chameleon change buttons after they were added.

**Comment by Yenwodyah** [comment]
Even less of a reason to stop nonantags from forging them, then.

**Comment by necromanceranne** [comment]
Finally.

**Comment by nero1024** [comment]
Why can't just anyone forge an ID with the agent card in the first place?

**Comment by HomocidlePancakes** [comment]
Because it’s supposed to be a camouflage item/a way for you to stealthy gain more access and the fact that sec can just use it hand to check kinda nullifies it’s whole purpose.

**Comment by anconfuzedrock** [comment]
Yen be like
"oh, the thing is broken because of a very minor, fixable issue? I better use that issue as an excuse to change the thing, instead of fixing the minor, fixable issue."
heck no, that's not an excuse or we wouldn't have metastation in the game because it has access issues with the virology front door

**Comment by Yenwodyah** [comment]
Not sure what you mean confuzed. If you're talking about the chameleon action buttons, that's unrelated to this PR.

**Comment by ATH1909** [comment]
While you're at it, could you fix this bug?

https://github.com/tgstation/tgstation/issues/44087

**Comment by Yenwodyah** [comment]
@ATH1909 This should fix it, but it won't fix the underlying problem, which is probably some antag spawner forgetting to set special_role.

**Comment by HomocidlePancakes** [comment]
Non-traitors being able to cham buttons on the IDs was an oversight when they were added

**Comment by Shadowflame909** [comment]
Do you think now that anyone can use it, it's no longer worth its current price?

**Comment by Yenwodyah** [comment]
@Shadowflame909 As long as you disguise it as something after you buy it, non-traitors still can't use it.

File: code/game/objects/items/cards_ids.dm [hunk 1]
```diff
@@ -304,28 +304,32 @@ update_label("John Doe", "Clowny")
...
-					registered_name = ""
-					return
-				assignment = u
-				update_label()
-				to_chat(user, "<span class='notice'>You successfully forge the ID card.</span>")
+		if(!(user.mind.special_role || anyone)) //Unless anyone is allowed, only syndies can use the card, to stop metagaming.
+			if(!registered_name) //If a non-syndie is the first to forge an unassigned agent ID, then anyone can forge it.
+				anyone = TRUE
+			else
+				..()
```
Comment (ShizCalev) [inline code/game/objects/items/cards_ids.dm]: return ..()

File: code/game/objects/items/cards_ids.dm [hunk 2]
```diff
@@ -304,28 +304,32 @@ update_label("John Doe", "Clowny")
...
+
+			var/u = copytext(sanitize(input(user, "What occupation would you like to put on this card?\nNote: This will not grant any access levels other than Maintenance.", "Agent card job assignment", "Assistant")as text | null),1,MAX_MESSAGE_LEN)
+			if(!u)
+				registered_name = ""
 				return
+			assignment = u
+			update_label()
+			to_chat(user, "<span class='notice'>You successfully forge the ID card.</span>")
+			return
 	..()
```
Comment (ShizCalev) [inline code/game/objects/items/cards_ids.dm]: return ..()

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Pull request merged**